### PR TITLE
Woo: Add product update page.

### PR DESCRIPTION
**Pull request by coderkevin**
This adds the product update editing page.

To Test:

1. `npm start`
2. Visit `http://calypso.localhost:3000/store/products/<site_url>`
3. Select a product from the list.
4. Observe that the edit page shows up, populated with data from the product you selected.

### CODE REVIEW

File: client/extensions/woocommerce/app/products/product-update.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+	onTrash = () => {
+		// TODO: Add action dispatch to trash this product.
+	}
+
+	onSave = () => {
+		const { product, translate } = this.props;
+
+		const successAction = successNotice(
+			translate( '%(product)s successfully updated.', {
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/products/product-update.js]: Hi! I've found a possible matching string that has already been translated **57** times:
`translate( '%1$s updated successfully.' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woo-product-update-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by kellychoffman** [comment]
This doesn't work consistently for me. Sometimes it linked me to a different product. On variable products, the Variations toggle wasn't in the on position. When I toggled it on, I could see the correct data, though.

**Comment by belcherj** [comment]
I had a bunch of issues where product didn't show up.

**Comment by belcherj** [comment]
screen shot 2017-06-28 at 2 10 51 pm

**Comment by coderkevin** [comment]
Yeah, there are issues where the page isn't updating correctly on startup. I'm on it. Thanks!

**Comment by coderkevin** [comment]
This PR should be working and needs a review.

File: client/extensions/woocommerce/app/products/product-update.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+import { editProductCategory } from 'woocommerce/state/ui/product-categories/actions';
+import { getActionList } from 'woocommerce/state/action-list/selectors';
+import { getCurrentlyEditingProduct } from 'woocommerce/state/ui/products/selectors';
+import { getProductVariationsWithLocalEdits } from 'woocommerce/state/ui/products/variations/selectors';
+import { editProductVariation } from 'woocommerce/state/ui/products/variations/actions';
+import { fetchProduct } from 'woocommerce/state/sites/products/actions';
+import { fetchProductVariations } from 'woocommerce/state/sites/product-variations/actions';
+import { fetchProductCategories } from 'woocommerce/state/sites/product-categories/actions';
+import { getProductCategoriesWithLocalEdits } from 'woocommerce/state/ui/product-categories/selectors';
+import { createProduct } from 'woocommerce/state/sites/products/actions';
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/products/product-update.js]: You are importing `fetchProduct` above from the same actions file -- so these can be combined.

File: client/extensions/woocommerce/app/products/product-update.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Main from 'components/main';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import { successNotice, errorNotice } from 'state/notices/actions';
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/products/product-update.js]: Some of these are out of alphabetical order. Like this, or create product.

File: client/extensions/woocommerce/app/products/product-update.js (hunk 2 above)
Comment (coderkevin) [inline client/extensions/woocommerce/app/products/product-update.js]: Thanks!

File: client/extensions/woocommerce/app/products/product-update.js (hunk 3 above)
Comment (coderkevin) [inline client/extensions/woocommerce/app/products/product-update.js]: Alphabetized.

File: client/extensions/woocommerce/app/products/product-header.js [hunk 4]
```diff
@@ -13,21 +13,51 @@ import ActionHeader from 'woocommerce/components/action-header';
...
+}
+
+function renderSaveButton( onSave, product, isBusy, translate ) {
 	const saveExists = 'undefined' !== typeof onSave;
 	const saveDisabled = false === onSave;
 
-	const saveButton = saveExists &&
+	const saveLabel = ( product && ! isObject( product.id )
+		? translate( 'Update' )
+		: translate( 'Save & Publish' )
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/products/product-header.js]: Hi! I've found a possible matching string that has already been translated **67** times:
`translate( 'Save &amp; Publish' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**